### ISSUE TITLE
Test that graph readers and writers are consistent

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#1922]
It would be good to have a set of tests that verify that graph writers and readers are consistent:

 - Create a graph (with attributes and vertex names).
 - Write it to a file.
 - Read it back from the file.
 - If the format does not naturally preserve the vertex ordering, then reorder vertices to have the vertex names in alphabetical order (`igraph_permute_vertices()`).
 - Compare that the graph read back from the file, as well as its attributes, are the same as the graph originally written out. See `igraph_is_same_graph()`, and when checking edge attributes, canonicalize edge ordering.

**Comment by szhorvat** [comment]
I suggest adding these tests on `develop` only. I made several robustness fixes on `develop` recently, and now it's difficult to port fixes between `master` and `develop`.

NCOL is a good one to start with because it is simple and has no tests so far. On `develop`, I changed the NCOL writer to preserve edge ordering. This can be tested too.

With formats that do not preserve the edge ordering, I think we can proceed like this in verifying that the edge attributes match: (1) match up vertex ordering between the graphs as discussed above (2) iterate through edges as `IGRAPH_EDGEORDER_FROM`.  I _think_ this would make the edge ordering predictable assuming that there are no multi-edges. If there are multi-edges, there's no way to match up edges between the two graphs other than through attributes.
